src/main/java/seedu/address/model/person/Attendance.java

```diff
@@ -0,0 +1,46 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a Person's attendance in the address book.
+ * Guarantees: immutable; is always valid
+ */
+public class Attendance {
+    protected static final DateTimeFormatter DTF = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    public final LocalDate time;
+
+    /**
+     * Constructs an {@code Attendance}.
+     *
+     * @param attendance a description of the attendance.
+     */
+    public Attendance(String attendance) {
+        requireNonNull(attendance);
+        this.time = LocalDate.parse(attendance, DTF);
```

Comment:
`this` keyword can be removed since there's no conflict in variable names.

Reply:
yeap thanks! cleaner this way